main.go

```diff
@@ -1,142 +1,34 @@
-	username string
-	apiKey   string
-}
-
-func createPage(spaceID, title, htmlPageContent, parentID string, credentials authenticationCredentials) (newPageId string, err error) {
-	url := fmt.Sprintf("https://%s/wiki/api/v2/pages", readConfluenceDomainFromEnvironmentVariable())
-
-	payload := CreatePageRequestBody{
-		SpaceID:  spaceID,
-		Status:   "current",
-		Title:    title,
-		ParentID: parentID,
-	}
-	payload.Body.Representation = "storage"
-	payload.Body.Value = htmlPageContent
-
-	jsonPayload, err := json.Marshal(payload)
-	if err != nil {
-		return "", fmt.Errorf("failed to marshal payload: %w", err)
-	}
-
-	req, err := http.NewRequest("POST", url, bytes.NewBuffer(jsonPayload))
-	if err != nil {
-		return "", fmt.Errorf("failed to create request: %w", err)
-	}
-
-	req.SetBasicAuth(credentials.username, credentials.apiKey)
-	req.Header.Set("Accept", "application/json")
-	req.Header.Set("Content-Type", "application/json")
-
-	client := &http.Client{Timeout: 10 * time.Second}
-	resp, err := client.Do(req)
-	if err != nil {
-		return "", fmt.Errorf("request failed: %w", err)
-	}
-	defer resp.Body.Close()
-
-	bodyBytes, err := io.ReadAll(resp.Body)
-	if err != nil {
-		return "", fmt.Errorf("failed to read response body: %w", err)
-	}
-
-	if resp.StatusCode != http.StatusOK {
-		return "", fmt.Errorf("error response code: %d; body: %s", resp.StatusCode, string(bodyBytes))
-	}
-
-	var response CreatePageResponseBodyStatusOk
-	if err := json.Unmarshal(bodyBytes, &response); err != nil {
-		return "", fmt.Errorf("failed to unmarshal response: %w", err)
-	}
-
-	return response.Id, nil
-}
-
-func readAuthenticationCredentialsFromEnvironmentVariables() authenticationCredentials {
-	return authenticationCredentials{
-		username: os.Getenv("CONFLUENCE_USERNAME"),
-		apiKey:   os.Getenv("CONFLUENCE_API_KEY"),
-	}
-}
-
-func readConfluenceDomainFromEnvironmentVariable() string {
-	return os.Getenv("CONFLUENCE_DOMAIN")
-}
-
-func convertMarkdownToHTML(markdownFilePath string) (html string, err error) {
-	file, err := os.Open(markdownFilePath)
-	if err != nil {
-		return "", fmt.Errorf("failed to open markdown file: %w", err)
-	}
-	defer file.Close()
-
-	content, err := io.ReadAll(file)
-	if err != nil {
-		return "", fmt.Errorf("failed to read markdown file: %w", err)
-	}
-
-	htmlContent := blackfriday.Run(content)
-	return string(htmlContent), nil
-}
-
 func main() {
-	var spaceID, markdownFilePath, title string
-	flag.StringVar(&spaceID, "spaceID", "", "The ID of the Confluence space where the page will be created.")
-	flag.StringVar(&markdownFilePath, "markdownFilePath", "./", "The path to the Markdown file you want to publish.")
-	flag.StringVar(&title, "title", "", "What you wish to title the generated Confluence page.")
+	var markdownFilePath, title string
+	flag.StringVar(&markdownFilePath, "markdownFilePath", "", "The path to the Markdown file you want to publish.")
 	flag.Parse()
 
+	title = filepath.Base(markdownFilePath)
```

Comment:
At first I wanted to do this too, but having spaces in the file name breaks relative path links. But if there's a way to make them work with spaces, this would be so much better! Would just want to trim the `.md`

Reply:
I'd rather not try to address this issue. 
The issue won't persist in our use case and for any hypothetical random user the api is still in tact since this is essentially "glue" code that they would need to go in and write themselves. 

I see your ask to slice the `.md` off and now I'm thinking about addressing edge cases and I don't wanna keep thinking about it lol.

Reply:
If you don't want to address it now, could we revert to the previous implementation of passing the title as an arg? I don't want to end up with ugly Confluence page titles that don't have spaces

Reply:
Addressed.

Reply:
The `formatBase` function doesn't seem like it's addressing the problem of user-friendly titles that don't have spaces

Reply:
Best I can do brthr.

Reply:
My `ProjectDirectoryLayout.md` file got published as "Projectdirectorylayout" 🥴 

I really don't think trying to auto-generate a user-friendly string from a non-user-friendly string is the way to go. I think there should just be a place where the user gets to explicitly state what they want a page to be titled. I feel like https://github.com/JackTimothy/pubdoc/issues/3 will be a good time to figure that out

Reply:
In `JackTimothy:main` there's already an implementation for pulling this from the command line. Obviously we won't have each individual page get published via a separate call to this script, but we'll likely end up having a function for publishing a page. That future function could take the title as a parameter

Reply:
Or maybe the title just comes from the first line of the Markdown file? Then that line gets trimmed before the Markdown is converted...

Reply:
Hold up: https://chatgpt.com/share/6769ec9a-589c-800f-87bc-aae45b612a64

Reply:
I just realized in your original reply you said "The issue won't persist in our use case". Maybe you know something I don't if you think this problem will solve itself. But in any case, turns out we can make relative path links work even when spaces are in the file name, so I think we could simply trim `.md` and use the file name without doing any other edits. We'll just have to take into account the workaround for spaces when we do https://github.com/JackTimothy/pubdoc/issues/4

Reply:
The %20 method is the only one that VS Code's Markdown preview seems to accept

Reply:
I'm familiar with the `\ ` method. 

Regardless when I said that "this won't be a problem in the future". I'm referring to either creating the file paths internal to the program, or 2 reading them from the filesystem. 

I'd much rather simply create them myself inside the program. 

My original intent with this idea was to define an authoritative config representation. (either json schema, or straight up golang structs I don't mind.) Then use those to "build" the citation heavy pages. 

Think reworking some of the big pages at work. 

In my imaginary software when updating SC the dev will simply go edit one field and then that will be distributed to all of the relevant confluence endpoints via this tool. 

So endgame since you want to represent everything as markdown I'm imagining. 

1. Generate SSOT. JSON or structs idc. 
2. Generate Markdown or HTML by using SSOT to disperse content. 
3. Publish to confluence. 

Then ENDGAME is this just happens on every commit and when you tag a version the full version docs get published under that tag.

Reply:
I'm unsure of how we would want to "source" the SSOT. 

Now that I'm aware that it appears that we can just use raw HTML. I think it'd be worth considering potentially just going straight to html with something like [this](https://github.com/tylermmorton/tmpl). Or some other version of html templating. 

It would also be a good idea to go straight to html if we wanted to abandon confluence... LOL

Reply:
Future work. 

The upside to abandoning confluence would be being allowed to do whatever the hell we want on the UI. I would sign up for that work if we were to do something like htmx, (_which is all you would need for this, but I'm aware I'm fighting an uphill battle trying to avoid a SPA JS framework_). 

I'm rambling. Just merge the PR lol.

Reply:
lol abandoning Confluence is above my paygrade rn